File: archive/p/php/file-input-output.php
<?php

function write_file($file_name)
{
    $file = @fopen($file_name, "w");
    if ($file === false) {
        echo "Cannot open file \"", $file_name, "\" for writing.\n";
        return false;
    }

    fwrite($file, "Hello World.\n");
    fwrite($file, "Testing one...\n");
    fwrite($file, "two...\n");
    fwrite($file, "three!\n");

    fflush($file);
    fclose($file);
    return true;
}

function read_file($file_name)
{
    $file = @fopen($file_name, "r");
    if ($file === false) {
        echo "Cannot open file \"", $file_name, "\" for reading.\n";
        return;
    }

    while (!feof($file)) {
        echo fgets($file);
    }

    fclose($file);
}

$file_name = "output.txt";
if (write_file($file_name)) {
    read_file($file_name);
}

File: archive/p/perl/file-input-output.pl
#!/usr/bin/env perl
use v5.42;
use IO::File;

my $file    = "output.txt";
my $content = <<'EOF';
Perl is cool!
There's more than one way to do it.
EOF

# Write to file

my $w = IO::File->new(">$file") or die "Cannot open $file for writing: $!";
$w->print($content)             or die "Write failed: $!";
$w->close                       or die "Cannot close after write: $!";

# Read from file

my $r = IO::File->new("<$file") or die "Cannot open $file for reading: $!";
print while <$r>;
$r->close or die "Cannot close after read: $!";
